### PR TITLE
"MysqlDatabase" "MysqlUser" Add delete policy

### PR DESCRIPTION
**Pull request by jiuker**
add
"MysqlDatabase" "MysqlUser" Add delete policy

Sometimes deleting these resources is very dangerous, so adding a certain label can reduce data loss

### CODE REVIEW

File: pkg/controller/mysqldatabase/db_controller.go [hunk 1]
```diff
@@ -94,9 +94,13 @@ func (r *ReconcileMySQLDatabase) Reconcile(ctx context.Context, request reconcil
 
 	// Check if the resource is deleted
 	if !db.ObjectMeta.DeletionTimestamp.IsZero() {
-		err = r.deleteDatabase(ctx, db)
-		if err != nil {
-			return reconcile.Result{}, err
+		if db.ObjectMeta.Labels != nil && mysqlv1alpha1.MysqlResourceDeletionPolicy(db.ObjectMeta.Labels[mysqlv1alpha1.MysqlResourceDeletionPolicyLabel]) == mysqlv1alpha1.MysqlResourceDeletionPolicyRetain {
```
Comment (cndoit18) [inline pkg/controller/mysqldatabase/db_controller.go:97]: This should be in the annotation, not in the label

Comment (jiuker) [reply pkg/controller/mysqldatabase/db_controller.go:97]: LGTM

Comment (jiuker) [reply pkg/controller/mysqldatabase/db_controller.go]: > This should be in the annotation, not in the label

New commit，review plz。

**Comment by jiuker** [comment]
review again。 @cndoit18

File: pkg/controller/mysqldatabase/db_controller.go [hunk 2]
```diff
@@ -94,9 +94,13 @@ func (r *ReconcileMySQLDatabase) Reconcile(ctx context.Context, request reconcil
 
 	// Check if the resource is deleted
 	if !db.ObjectMeta.DeletionTimestamp.IsZero() {
-		err = r.deleteDatabase(ctx, db)
-		if err != nil {
-			return reconcile.Result{}, err
+		if db.ObjectMeta.Annotations != nil && mysqlv1alpha1.MysqlResourceDeletionPolicy(db.ObjectMeta.Annotations[mysqlv1alpha1.MysqlResourceDeletionPolicyAnnotationKey]) == mysqlv1alpha1.MysqlResourceDeletionPolicyRetain {
```
Comment (cndoit18) [inline pkg/controller/mysqldatabase/db_controller.go:97]: you can add a helper function to simplify this part

Comment (jiuker) [reply pkg/controller/mysqldatabase/db_controller.go]: yes ,LGTM

**Comment by jiuker** [comment]
review again。 @cndoit18

**Review by cndoit18 — APPROVED** [review]
LGTM

**Comment by jiuker** [comment]
Is there anything else to deal with? When can it be closed?

**Comment by cndoit18** [comment]
> Is there anything else to deal with? When can it be closed? @cndoit18

need to wait for CI to pass

**Comment by jiuker** [comment]
> > Is there anything else to deal with? When can it be closed? @cndoit18
> 
> need to wait for CI to pass

It looks like lint failed. Do I need to fix it? @cndoit18

**Comment by jiuker** [comment]
@cndoit18 All ci passed.review again plz.

**Comment by cndoit18** [comment]
Hi, please add unit test

**Comment by jiuker** [comment]
unit test is ready.review plz @cndoit18

**Review by cndoit18 — APPROVED** [review]
LGTM

**Comment by jiuker** [comment]
Does it look conflicting? @cndoit18

**Comment by cndoit18** [comment]
> Does it look conflicting? @cndoit18 
> 
> 
hi, squash your commits, thx

**Pull request closed without merging**